c_src/image/cairo_image_ops.c

```diff
@@ -0,0 +1,66 @@
+#include <arpa/inet.h>
+#include <cairo.h>
+#include <stdlib.h>
+
+#include "cairo_ctx.h"
+#include "comms.h"
+#include "image_ops.h"
+
+uint32_t convert_rgba_to_argb(uint32_t pixel)
+{
+  return ((pixel & 0xFFFFFF00) >> 8) | ((pixel & 0xFF) << 24);
+}
+
+int32_t image_ops_create(void* v_ctx,
+                         uint32_t width, uint32_t height,
+                         void* p_pixels)
+{
+  cairo_format_t format = CAIRO_FORMAT_ARGB32;
+  int stride = cairo_format_stride_for_width(format, width);
+  size_t num_pixels = width * height;
+  uint32_t* rgba_pixels = (uint32_t*)p_pixels;
+  uint32_t* argb_pixels = malloc(num_pixels * sizeof(uint32_t));
```

Comment:
⚠️ (Recommended) We should be checking for `NULL`s here and other `*alloc` places. We are dealing w/ devices w/ fixed resources.

P.S. This can be handled in another PR, just make sure it gets in the backlog.

Reply:
I'm added a check for NULL here to return 0 as `script_ops_fill_image()` that then use the image_id generated from here, check if it is NULL